### PR TITLE
Added Installation, Gradient Accumulation and Data Iterator How To's from FAQs

### LINKED ISSUE #1: Create How-To-Guides notebooks
Shift the files from pytorch-ignite/pytorch-ignite.github.io/content/docs/how-to-guides to pytorch-ignite/examples/how-to-guides and store them as .ipynb instead of .md

### PR DESCRIPTION
**Pull request by Priyansi**
Fixes #1 

cc @trsvchn @sdesrozis @vfdev-5

### CODE REVIEW

File: how-to-guides/installation.ipynb [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    "|  NVIDIA Apex  |        [`pytorchignite/apex:latest`](https://hub.docker.com/r/pytorchignite/apex)        |        [`pytorchignite/hvd-apex:latest`](https://hub.docker.com/r/pytorchignite/hvd-apex)        |        [`pytorchignite/msdp-apex:latest`](https://hub.docker.com/r/pytorchignite/msdp-apex)        |\n",
+    "| Apex + Vision | [`pytorchignite/apex-vision:latest`](https://hub.docker.com/r/pytorchignite/apex-vision) | [`pytorchignite/hvd-apex-vision:latest`](https://hub.docker.com/r/pytorchignite/hvd-apex-vision) | [`pytorchignite/msdp-apex-vision:latest`](https://hub.docker.com/r/pytorchignite/msdp-apex-vision) |\n",
+    "|  Apex + NLP   |    [`pytorchignite/apex-nlp:latest`](https://hub.docker.com/r/pytorchignite/apex-nlp)    |    [`pytorchignite/hvd-apex-nlp:latest`](https://hub.docker.com/r/pytorchignite/hvd-apex-nlp)    |    [`pytorchignite/msdp-apex-nlp:latest`](https://hub.docker.com/r/pytorchignite/msdp-apex-nlp)    |\n",
+    "\n",
+    "and run it with Docker v19.03+ :\n",
+    "\n",
+    "``` shell\n",
+    "docker run --gpus all -it -v $PWD:/workspace/project --network=host --shm-size 16G IMAGE_NAME\n",
```
Comment (sdesrozis) [inline how-to-guides/installation.ipynb:72]: What do you think to add an example to run one of the usecase of the database ? (cifar, voc, whatever)

**Review by sdesrozis — COMMENTED** [review]
@vfdev-5 mentioned on discord this repo
* https://github.com/SapienzaNLP/gsrl/blob/0e3cb86c1669c3fc4407d4080a0095060feaaaae/src/bin/train.py#L99-L106
* https://github.com/SapienzaNLP/gsrl/blob/0e3cb86c1669c3fc4407d4080a0095060feaaaae/src/bin/train.py#L123-L131

Maybe we could improve this notebook to have a better split approach ?

File: how-to-guides/installation.ipynb (hunk 1 above)
Comment (Priyansi) [reply how-to-guides/installation.ipynb:72]: Do you mean a tutorial on pulling a docker image and then training a basic model on any popular dataset?

Comment (sdesrozis) [reply how-to-guides/installation.ipynb:72]: Yes, using an example installed in the images.

File: how-to-guides/data-iterator.ipynb [hunk 2]
```diff
@@ -0,0 +1,624 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "80e1c34f-37b9-4304-bddd-12e5b2fbcc63",
+   "metadata": {},
+   "source": [
+    "# How to work with data in the form of iterators\n",
```
Comment (vfdev-5) [inline how-to-guides/data-iterator.ipynb:8]:
```suggestion
    "# How to work with data iterators ?\n",
```

File: how-to-guides/data-iterator.ipynb [hunk 3]
```diff
@@ -0,0 +1,624 @@
...
+    "# We need to specify epoch_length to define the epoch\n",
+    "trainer.run(infinite_iterator(4), epoch_length=5, max_epochs=3)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cb9743c9-1fc0-4855-8a45-a7f7242118b9",
+   "metadata": {},
+   "source": [
+    "If we do not specify **epoch_length**, we can stop the training explicitly by calling `~ignite.engine.engine.Engine.terminate` In this case, there will be only a single epoch defined."
```
Comment (vfdev-5) [inline how-to-guides/data-iterator.ipynb:102]: Wrong link as `~ignite.engine.engine.Engine.terminate`, please fix

File: how-to-guides/data-iterator.ipynb [hunk 4]
```diff
@@ -0,0 +1,624 @@
...
+   "id": "aa425c90-da2e-49e9-aec7-f45000ec32cc",
+   "metadata": {},
+   "source": [
+    "### Finite iterator with unknown length\n",
+    "\n",
+    "Let's use a finite data iterator but with unknown length (for user). In\n",
+    "case of training, we would like to perform several passes over the\n",
+    "dataflow and thus we need to restart the data iterator when it is\n",
+    "exhausted. In the code, we do not specify <span\n",
+    "class=\"title-ref\">epoch_length</span> which will be automatically\n",
```
Comment (vfdev-5) [inline how-to-guides/data-iterator.ipynb:199]: Fix `epoch_length`

File: how-to-guides/data-iterator.ipynb [hunk 5]
```diff
@@ -0,0 +1,624 @@
...
+   "id": "446babd9-d6f5-4407-be8a-0224b8b0564f",
+   "metadata": {},
+   "source": [
+    "### Finite iterator with known length\n",
+    "\n",
+    "Let's use a finite data iterator with known size for training or\n",
+    "validation. If we need to restart the data iterator, we can do this\n",
+    "either as in case of unknown size by attaching the restart handler on\n",
+    "<span\n",
+    "class=\"title-ref\">@trainer.on(Events.DATALOADER_STOP_ITERATION)</span>,\n",
```
Comment (vfdev-5) [inline how-to-guides/data-iterator.ipynb:404]: FIx `@trainer.on(Events.DATALOADER_STOP_ITERATION)` rendering

File: how-to-guides/data-iterator.ipynb [hunk 6]
```diff
@@ -0,0 +1,624 @@
...
+   ],
+   "source": [
+    "import torch\n",
+    "from ignite.engine import Engine, Events\n",
+    "\n",
+    "torch.manual_seed(12)\n",
+    "\n",
+    "def finite_unk_size_data_iter():\n",
+    "    for i in range(11):\n",
+    "        yield i\n",
```
Comment (trsvchn) [inline how-to-guides/data-iterator.ipynb:299]: I think this can be simplified with `yield from`:
```python
def finite_unk_size_data_iter():
     yield from range(11)
```

File: how-to-guides/data-iterator.ipynb [hunk 7]
```diff
@@ -0,0 +1,624 @@
...
+   "source": [
+    "import torch\n",
+    "from ignite.engine import Engine, Events\n",
+    "\n",
+    "torch.manual_seed(12)\n",
+    "\n",
+    "def infinite_iterator(batch_size):\n",
+    "    while True:\n",
+    "        batch = torch.rand(batch_size, 3, 32, 32)\n",
+    "        yield batch\n",
```
Comment (trsvchn) [inline how-to-guides/data-iterator.ipynb:82]: What about to use `count` from `itertools`?
cause in the next section `Finite iterator with unknown length` we actually do counting with `range(11)`

**Review by trsvchn — COMMENTED** [review]
Thanks Priyansi! Good work! I left some comments aiming to reduce the amount of non-ignitre code.
@vfdev-5 can we use `itertools.count` here as a infinite generator?

**Comment by vfdev-5** [comment]
> can we use itertools.count in how-to-guides/data-iterator.ipynb as a infinite generator?

@trsvchn initially all these methods like `infinite_iterator`, `finite_unk_size_data_iter` are just to emulate data iterator and not intended to show a best practice of how to code them, but just to showcase that ignite can work with them. 
I was thinking that using other tools like itertools can make the code more complicated to understand.

**Comment by trsvchn** [comment]
@vfdev-5 Thanks! Agreed! Let's not complicate things then.

**Comment by vfdev-5** [comment]
@Priyansi I left few comments to fix. Let's finalize this PR today

**Review by vfdev-5 — APPROVED** [review]
Thanks @Priyansi
